subsys/usb/usb_c/usbc_pe_snk_states.c

```diff
@@ -0,0 +1,839 @@
+				break;
+			default:
+				pe_set_state(dev, PE_SEND_NOT_SUPPORTED);
+			}
+			return;
+		}
+		/* Control Messages */
+		else {
+			switch (header.message_type) {
+			case PD_CTRL_GOOD_CRC:
+				/* Do nothing */
+				break;
+			case PD_CTRL_PING:
+				/* Do nothing */
+				break;
+			case PD_CTRL_GET_SINK_CAP:
+				pe_set_state(dev, PE_SNK_GIVE_SINK_CAP);
+				return;
+			case PD_CTRL_DR_SWAP:
+				pe_set_state(dev, PE_DRS_EVALUATE_SWAP);
+				return;
+			case PD_CTRL_NOT_SUPPORTED:
+				/* Do nothing */
+				break;
+			/*
+			 * USB PD 3.0 6.8.1:
+			 * Receiving an unexpected message shall be responded
+			 * to with a soft reset message.
+			 */
+			case PD_CTRL_ACCEPT:
+			case PD_CTRL_REJECT:
+			case PD_CTRL_WAIT:
+			case PD_CTRL_PS_RDY:
+				pe_send_soft_reset(dev, prl_rx->emsg.type);
+				return;
+			/*
+			 * Receiving an unknown or unsupported message
+			 * shall be responded to with a not supported message.
+			 */
+			default:
+				pe_set_state(dev, PE_SEND_NOT_SUPPORTED);
+				return;
+			}
+		}
+	}
+
+	/*
+	 * Check if we are waiting to resend any messages
+	 */
+	if (usbc_timer_expired(&pe->pd_t_wait_to_resend)) {
+		if (atomic_test_and_clear_bit(&pe->flags, PE_FLAGS_WAIT_SINK_REQUEST)) {
+			pe_set_state(dev, PE_SNK_SELECT_CAPABILITY);
+			return;
+		} else if (atomic_test_and_clear_bit(&pe->flags, PE_FLAGS_WAIT_DATA_ROLE_SWAP)) {
+			pe_set_state(dev, PE_DRS_SEND_SWAP);
+			return;
+		}
+	}
+
+	/*
+	 * Handle Device Policy Manager Requests
+	 */
+	sink_dpm_requests(dev);
+}
+
+/**
+ * @brief PE_SNK_Hard_Reset Entry State
+ */
+void pe_snk_hard_reset_entry(void *obj)
+{
+	struct policy_engine *pe = (struct policy_engine *)obj;
+	const struct device *dev = pe->dev;
+	struct usbc_port_data *data = dev->data;
+
+	LOG_INF("PE_SNK_Hard_Reset");
+
+	/*
+	 * Note: If the SinkWaitCapTimer times out and the HardResetCounter is
+	 *       greater than nHardResetCount the Sink Shall assume that the
+	 *       Source is non-responsive.
+	 */
+	if (atomic_test_bit(&pe->flags, PE_FLAGS_SNK_WAIT_CAP_TIMEOUT) &&
+	    pe->hard_reset_counter > PD_N_HARD_RESET_COUNT) {
+		/* Inform the DPM that the port partner is not responsive */
+		policy_notify(dev, PORT_PARTNER_NOT_RESPONSIVE);
+
+		/* Pause the Policy Engine */
+		data->pe_enabled = false;
+		return;
+	}
+
+	atomic_clear_bit(&pe->flags, PE_FLAGS_SNK_WAIT_CAP_TIMEOUT);
+	atomic_clear_bit(&pe->flags, PE_FLAGS_PROTOCOL_ERROR);
+
+	/* Request the generation of Hard Reset Signaling by the PHY Layer */
+	prl_execute_hard_reset(dev);
+	/* Increment the HardResetCounter */
+	pe->hard_reset_counter++;
+}
+
+/**
+ * @brief PE_SNK_Hard_Reset Run State
+ */
+void pe_snk_hard_reset_run(void *obj)
+{
+	struct policy_engine *pe = (struct policy_engine *)obj;
+	const struct device *dev = pe->dev;
+
+	/*
+	 * Transition to the PE_SNK_Transition_to_default state when:
+	 *  1) The Hard Reset is complete.
+	 */
+	if (atomic_test_bit(&pe->flags, PE_FLAGS_HARD_RESET_PENDING)) {
```

Comment:
I think I missed where this bit gets set in the first place?

Reply:
I added code to set the bit in the entry function